pkg/hubble/observer/namespace_manager_test.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package observer
+
+import (
+	"testing"
+	"time"
+
+	observerpb "github.com/cilium/cilium/api/v1/observer"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNamespaceManager(t *testing.T) {
+	// mock time
+	currentTime := time.Time{}
+	mgr := NewNamespaceManager()
+	// use the mocked time
+	mgr.nowFunc = func() time.Time {
+		return currentTime
+	}
+	advanceTime := func(d time.Duration) {
+		// update our currentTime
+		currentTime = currentTime.Add(d)
+		// trigger cleanupNamespaces after we advance time to ensure it's run
+		mgr.cleanupNamespaces()
+	}
+
+	// we start with no namespaces
+	expected := []*observerpb.Namespace{}
+	assert.Equal(t, expected, mgr.GetNamespaces())
+
+	// add a few namespaces
+	nses1 := []*observerpb.Namespace{
+		{Namespace: "ns-2"},
+		{Namespace: "ns-1"}, // out of order, we'll verify it's sorted when we call GetNamespaces later
+	}
+	mgr.AddNamespace(nses1...)
+
+	// namespaces that we added should be returned, sorted
+	expected = []*observerpb.Namespace{
+		{Namespace: "ns-1"},
+		{Namespace: "ns-2"},
+	}
+	assert.Equal(t, expected, mgr.GetNamespaces())
+
+	// advance the clock by 30 minutes and verify our namespaces are still known (TTL is 1 hour)
+	advanceTime(30 * time.Minute)
```

Comment:
nit: could use `namespaceTTL / 2` here instead, so that updating the TTL won't break the test.